src/facade/dragonfly_connection.cc

```diff
@@ -316,6 +314,34 @@ QueueBackpressure& GetQueueBackpressure() {
 
 thread_local vector<Connection::PipelineMessagePtr> Connection::pipeline_req_pool_;
 
+class PipelineCacheSizeTracker {
+ public:
+  bool CheckAndUpdateWatermark(size_t pipeline_sz) {
+    const auto now = Clock::now();
+    const auto elapsed = now - last_check_;
+    min_ = std::min(min_, pipeline_sz);
+    if (elapsed < std::chrono::milliseconds(10)) {
+      return false;
+    }
+
+    const bool watermark_reached = (min_ > 0);
+    min_ = Limits::max();
+    last_check_ = Clock::now();
+
+    return watermark_reached;
+  }
+
+ private:
+  using Tp = std::chrono::time_point<std::chrono::system_clock>;
+  using Clock = std::chrono::system_clock;
```

Comment:
Do we require millisecond resolution here or second resolution will also work? 
if you need milliseconds - then this code is buggy. I personally do not like std clocks because of this layer of abstractions, that make things more confusing. 
Also, please do not change to high resolution clock that is inefficient.

Reply:
> if you need milliseconds - then this code is buggy. 

In what sense this code is buggy if we need milliseconds ? The idea of `time points` expressed via `std::ratio` is that you get conversions almost for free between (ms, ns etc) ?